apps/course-api/src/controllers/courses.controller.ts

```diff
@@ -1,8 +1,10 @@
 import { Request, Response } from "express";
 import { paginate } from "@utils/pagination";
 import { getAllCourses, getCourseByCode } from "./get-courses";
+import { getRedisClient } from "@utils/redis_utils";
+
+const cacheTTL = 60 * 60 * 7; // cache will expire after 1 week
```

Comment:
do i also need a TTL? on studyplans

Reply:
hm.. I think we can either:

- Set a TTL for each endpoint
- or make it stays there unless is removed by the admin panel every time a new resource is uploaded

What do you think?

Reply:
or actually we can do both.. no harms anyways?

set TTL for each endpoint + have it deleted whenever a new resource is uploaded

Reply:
Yes sirr! I'll set it for maybe 7 days too or 30 days.